sql/connect/server/src/main/scala/org/apache/spark/sql/connect/execution/ExecuteThreadRunner.scala

```diff
@@ -311,6 +313,17 @@ private[connect] class ExecuteThreadRunner(executeHolder: ExecuteHolder) extends
     planner.process(command = command, responseObserver = responseObserver)
   }
 
+  private def handleMLCommand(request: proto.ExecutePlanRequest): Unit = {
```

Comment:
If we define the ML Command as a "kind" of command this becomes a bit easier and less burden on the ML side to manage the execution independently. You get all of the benefit of the infrastructure and it just works.